File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.py
class Solution:
    def findThePrefixCommonArray(self, A: List[int], B: List[int]) -> List[int]:
        res = []
        commCount = 0
        seen = [0 for _ in range(len(A))]
        for i in range(len(A)):
            seen[A[i] - 1] += 1
            if seen[A[i] - 1] == 2:
                commCount += 1
            seen[B[i] - 1] += 1
            if seen[B[i] - 1] == 2:
                commCount += 1
            res.append(commCount)
        return res

File: 2657-find-the-prefix-common-array-of-two-arrays/2657-find-the-prefix-common-array-of-two-arrays.cpp
class Solution {
public:
    vector<int> findThePrefixCommonArray(vector<int>& A, vector<int>& B) {
        vector<int> res;    
        int commCount = 0;
        vector<char> seen(A.size(), 0);
        for(int i = 0; i < A.size(); i++)
        {
            seen[A[i] - 1]++;
            if(seen[A[i] - 1] == 2)
            {
                commCount++;
            }
            seen[B[i] - 1]++;
            if(seen[B[i] - 1] == 2)
            {
                commCount++;
            }
            res.push_back(commCount);
        }
        return res;
    }
};